internal/subscriber/subscriber.go

```diff
@@ -541,9 +554,18 @@ func (s *Subscriber) logInMemoryStats() {
 		}
 		st[tp.String()] = total
 	}
+	logger.Ctx(s.ctx).Infow("subscriber: in-memory stats", s.getLogFields(), "stats", st)
+}
 
-	logger.Ctx(s.ctx).Infow("subscriber: in-memory stats", "stats", st,
-		"topic", s.topic, "subscription", s.subscription, "subscriberId", s.subscriberID)
+// returns a map of common fields to be logged
+func (s *Subscriber) getLogFields() map[string]interface{} {
```

Comment:
can we set them in context ?

Reply:
Had tried this. Values set in `ctx()` were not getting goroutine scoped. eg: `msgID` added to `ctx` of one request as getting used up in consecutive requests as well. Maybe I was doing it wrong. Looking for implementation suggestions using this.